substrate/frame/revive/src/gas.rs

```diff
@@ -219,16 +218,13 @@ impl<T: Config> GasMeter<T> {
 		Ok(ChargedAmount(amount))
 	}
 
-	/// Charge the specified amount of EVM gas.
-	/// This is used for basic opcodes (e.g arithmetic, bitwise, ...) that don't have a dedicated
-	/// benchmark
-	pub fn charge_evm_gas(&mut self, gas: u64) -> Result<(), DispatchError> {
-		let base_cost = T::WeightInfo::evm_opcode(1).saturating_sub(T::WeightInfo::evm_opcode(0));
-		self.gas_left = self
-			.gas_left
-			.checked_sub(&base_cost.saturating_mul(gas))
-			.ok_or_else(|| Error::<T>::OutOfGas)?;
-		Ok(())
+	/// Charge the specified token amount of gas or halt if not enough gas is left.
```

Comment:
Isn't that already the behavior of `charge`?

Reply:
yes but instruction function have this signature

```rust
pub fn push0<E: Ext>(interpreter: &mut Interpreter<E>) -> ControlFlow<Halt> {
	interpreter.ext.charge_or_halt(EVMGas(BASE))?;
	interpreter.stack.push(U256::zero())
}
```
so it's handy to have something returning a ControlFlow so that you can just use `?` inside the function.

I could  move that on the interpreter and do 
```rust
	interpreter.charge_or_halt(EVMGas(BASE))?;
```

but then you run into borrowing issue if we take a &mut on some other field like the stack